src/umpire/Umpire.hpp

```diff
@@ -189,6 +192,28 @@ MPI_Comm get_communicator_for_allocator(Allocator a, MPI_Comm comm);
 void register_external_allocation(void* ptr, util::AllocationRecord record);
 util::AllocationRecord deregister_external_allocation(void* ptr);
 
+/*!
+ * \brief Returns the Camp resource associated with a particular allocation
+ * **This function is used mainly for testing purposes.**
+ *
+ * \param Umpire allocator which was used to allocate the data
+ * \param Pointer which was used for the allocation
+ *
+ * \return Camp resource associated with the allocation *assuming the Allocator
+ * passed in is a ResourceAwarePool strategy and the allocation is either used or pending*
+ */
+Resource getResource(Allocator a, void* ptr);
+
+/*!
+ * \brief Returns the number of pending chunks associated with a ResourceAwarePool Allocator
+ * **This function is used mainly for testing purposes.**
+ *
+ * \param Umpire ResourceAwarePool allocator
+ *
+ * \return Number of currently pending chunks in the ResourceAwarePool
+ */
+std::size_t getNumPending(Allocator a);
```

Comment:
If this is for testing, can it not me a private member function of the ResourceAwarePool?

Reply:
I don't think I understand your question. Right now that is a protected member function of the RAP - I don't know why this function would be useful other than for testing purposes.. thoughts?

Reply:
This is also part of the issue for optimizations...